### ISSUE TITLE
paginator adapter querybuilder generate wrong count sql

### ISSUE THREAD

**Issue by zhao5908** [phalcon/cphalcon#9939]
I found this line of code generate wrong SQL

https://github.com/phalcon/cphalcon/blob/1.3.5/ext/paginator/adapter/querybuilder.c#L369

From version 1.3.3 began to have problems

when version 1.3.2 is 
> SELECT COUNT(*) AS `rowcount` FROM `tablename` WHERE ...

since 1.3.3
> SELECT COUNT(*) "rowcount" FROM (SELECT tablename.* FROM `tablename` WHERE (...) AS T

**Comment by dreamsxin** [comment]
Please submit the test code?

**Comment by zhao5908** [comment]
for example use https://github.com/phalcon/invo database and code

```php
        /* @var $modelsManager \Phalcon\Mvc\Model\Manager */
        $modelsManager = \Phalcon\DI::getDefault()->getShared('modelsManager');

        /* @var $builder \Phalcon\Mvc\Model\Query\BuilderInterface */
        $builder = $modelsManager->createBuilder();
        $builder->from('Products')->orderBy('id DESC');
        $builder->where('price < 100');
        $builder->columns('*');

        $paginator = new \Phalcon\Paginator\Adapter\QueryBuilder([
            "builder" => $builder,
            "limit" => 3,
            "page" => 1
        ]);

        $page = $paginator->getPaginate();
```

while the SQL is

```sql
    SELECT COUNT(*) "rowcount" FROM (SELECT products.* FROM `products` WHERE `products`.`price` < 100) AS T
```

for some reason, my table name is order, so order must add `, to prevent the keyword conflict, like this

    `order`.*

just like this

```sql
    SELECT COUNT(*) "rowcount" FROM (SELECT `products`.* FROM `products` WHERE `products`.`price` < 100) AS T
```

even so it still have a problem, when row count greater than 1000, will be very slow.

my development environment

```sql
    -- rowcount = 51817 , time = 550 ms
    SELECT COUNT(*) "rowcount" FROM (SELECT `order`.* FROM `order`) AS T;
    -- but
    -- rowcount = 51817 , time = 10 ms
    SELECT COUNT(*) "rowcount" FROM `order`;
```

thank you.

**Comment by dreamsxin** [comment]
First will fix about escape table name.
The sencond, you can see https://github.com/phalcon/cphalcon/issues/2065

**Comment by zhao5908** [comment]
thank you ,but still failed.

**Comment by dreamsxin** [comment]
It's not merge, you can pull branch https://github.com/dreamsxin/cphalcon/tree/issue_9939, test

**Comment by zhao5908** [comment]
tested 7bfe3a2 and issue_9939, but stil failed.

php -i
```
phalcon

Phalcon Framework => enabled
Phalcon Version => 1.3.5

Directive => Local Value => Master Value
phalcon.db.escape_identifiers => On => On
phalcon.orm.column_renaming => On => On
phalcon.orm.enable_literals => On => On
phalcon.orm.events => On => On
phalcon.orm.exception_on_failed_save => Off => Off
phalcon.orm.not_null_validations => On => On
phalcon.orm.virtual_foreign_keys => On => On
phalcon.register_psr3_classes => Off => Off
```

php -v
```
PHP 5.5.22 (cli) (built: Mar 10 2015 14:17:46)
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2015 Zend Technologies
    with Xdebug v2.2.7, Copyright (c) 2002-2015, by Derick Rethans
```

PHQL log
```
SELECT COUNT(*) "rowcount" FROM (SELECT order.* FROM `order` WHERE (`order`.`order_state` > :cancal) AS T
```

**Comment by dreamsxin** [comment]
You can add test code in file `ext/paginator/adapter/querybuilder.c`:
```c
PHALCON_CONCAT_VS(select_column, sql_table, ".*");
zend_printf("test:");
zend_print_zval_r(select_column, 0);
```

**Comment by andresgutierrez** [comment]
Could you please try again using Phalcon 2?

**Comment by max-mi** [comment]
I have the same issue with Phalcon 2.0.0/2.0.1

**Comment by andresgutierrez** [comment]
This is fixed in Phalcon 2.0.2/2.0.3, @max-mi Could you please try again using these versions?

**Comment by max-mi** [comment]
@andresgutierrez thank you for information. I've already simplified paginator query with automated calculations via triggers, but I will test this issue on newest Phalcon versions and let you know about results.
